hack/gha-reversemap.sh

```diff
@@ -0,0 +1,290 @@
+_get_sha_from_reversemap() {
+    action_ref=$1
+    query=$(yq ".\"$action_ref\".sha" $REVERSEMAP_FILE)
+    if [[ $? -ne 0 ]]; then
+        _exit_with_error "$query"
+    fi
+    _return "$query"
+}
+
+# Apply the commit sha of every actions listed in the reversemap as the version to use in all github workflows
+run_apply_reversemap() {
+    files=$@
+    for file in $files; do
+        _loginfo "applying '$REVERSEMAP_FILE' commit sha to be used in '$file' ..."
+        action_fullrefs=$(yq '.jobs[].steps[] | select(.uses) | .uses' "$file")
+        for action_fullref in $action_fullrefs; do
+            action_ref=$(echo "$action_fullref" | cut -d "@" -f 1)
+            action_sha=$(_get_sha_from_reversemap "$action_ref")
+            _loginfo "found '$action_ref' in reversemap with sha=$action_sha"
+            _update_action_version_infile "$file" "$action_ref" "$action_sha"
+        done
+    done
+}
+
+# Update the version of the given action references within the reverse map (sha, tag, urls) to its latest regular release tag
+run_update_action_version() {
+    action_refs=$@
+    for action_ref in $action_refs; do
+        _loginfo "updating dependency '$action_ref' tag to latest version available inside reverse map '$REVERSEMAP_FILE'"
+        latest_tag=$(_fetch_latest_tag "$action_ref")
+        action_sha=$(_fetch_sha_from_upstream_ref "$action_ref" "$latest_tag")
+        _yq_update_reversemap "$action_ref" "$latest_tag" "$action_sha"
+    done
+}
+
+# Update the reverse map values (sha, tag, urls) from the actions used in given github workflows
+run_update_reversemap() {
+    files=$@
+    for file in $files; do
+        _loginfo "updating $REVERSEMAP_FILE with actions from '$file' ..."
+        _update_reversemap_with "$file"
+    done
+}
+
+# Run the CLI with its operations
+run_cli() {
+    operation_arg=$1
+    case $operation_arg in
+    "apply-reversemap")
+        shift
+        files=$@
+        if [[ "$#" -eq 0 ]]; then
+            files="$GITHUB_WORKFLOWS_PATH/*.yml"
+        fi
+        _loginfo "running $operation_arg on $files"
+        run_apply_reversemap $files
+        ;;
+    "update-action-version")
+        shift
+        action_refs=$@
+        if [[ "$#" -eq 0 ]]; then
+            _exit_with_error 1 "missing action ref to update. Format must be '{gh_owner}/{gh_repo}'"
+        fi
+        _loginfo "running $operation_arg on $action_refs"
+        run_update_action_version $action_refs
+        ;;
+    "update-reversemap")
+        shift
+        files=$@
+        if [[ "$#" -eq 0 ]]; then
+            files="$GITHUB_WORKFLOWS_PATH/*.yml"
+        fi
+        _loginfo "running $operation_arg on $files"
+        run_update_reversemap $files
+        ;;
+    *)
+        help
+        exit 1
+        ;;
+    esac
+
+}
+
+# Execute the main program
+main() {
+    if [[ -z $GITHUB_TOKEN ]]; then
+        _exit_with_error $ERR_GITHUB_TOKEN_INVALID "environment variable GITHUB_TOKEN is not set."
```

Comment:
This seems like an unnecessary restriction. One or more invocations could be done as an unauthenticated client, depending on the details of the invocations. Not a blocker.